cutil/initapi/validate_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initapi
+
+import (
+	"github.com/kris-nova/kubicorn/apis/cluster"
+	"testing"
+)
+
+func TestServerPoolCountsHappy(t *testing.T) {
+	c := &cluster.Cluster{
+		Name: "c",
+		ServerPools: []*cluster.ServerPool{
+			{
+				Name:     "p",
+				MaxCount: 1,
+				MinCount: 1,
+			},
+		},
+	}
+	err := serverPoolCounts(c)
+	if err != nil {
+		t.Fatalf("error message incorrect for valid server pool counts"+
+			"should be: nil\n"+
+			"got:       %v\n", err)
+	}
+}
+
+func TestServerPoolCountsSad(t *testing.T) {
```

Comment:
Following the syntax above, can we have matching test functions for each of the validate functions and not lump them together? I know it seems like a lot of work, but once we have the validation system in place, the function name will serve as the message that tells us what went wrong 😉